src/operator/do.ts

```diff
@@ -4,14 +4,50 @@ import {Observable} from '../Observable';
 import {PartialObserver} from '../Observer';
 
 /**
- * Returns a mirrored Observable of the source Observable, but modified so that the provided Observer is called
- * for every item emitted by the source.
- * This operator is useful for debugging your observables for the correct values or performing other side effects.
- * @param {Observer|function} [nextOrObserver] a normal observer callback or callback for onNext.
- * @param {function} [error] callback for errors in the source.
- * @param {function} [complete] callback for the completion of the source.
- * @reurns {Observable} a mirrored Observable with the specified Observer or callback attached for each item.
+ * Perform a side effect for every emission on the source Observable, but return
+ * an Observable that is identical to the source.
+ *
+ * <span class="informal">Intercepts each emission on the source and runs a
+ * function, but returns an output which is identical to the source.</span>
```

Comment:
Sure. I can add that somewhere, but I try to keep the `informal` description short (my rule of thumb so far is max 2 lines), so your suggestion is longer. I can put your sentence outside of this informal block, like under it. But in general the `informal` description is what I **verbally** explain to a work colleague, written down unaltered.